rust/operator-binary/src/kerberos.rs

```diff
@@ -0,0 +1,110 @@
+        source: SecretOperatorVolumeSourceBuilderError,
+    },
+}
+
+pub fn add_kerberos_pod_config(
+    hive: &HiveCluster,
+    role: &HiveRole,
+    cb: &mut ContainerBuilder,
+    pb: &mut PodBuilder,
+) -> Result<(), Error> {
+    if let Some(kerberos_secret_class) = hive.kerberos_secret_class() {
+        // Mount keytab
+        let kerberos_secret_operator_volume =
+            SecretOperatorVolumeSourceBuilder::new(kerberos_secret_class)
+                .with_service_scope(hive.name_any())
+                .with_kerberos_service_name(role.kerberos_service_name())
+                .build()
+                .context(AddKerberosSecretVolumeSnafu)?;
+        pb.add_volume(
+            VolumeBuilder::new("kerberos")
+                .ephemeral(kerberos_secret_operator_volume)
+                .build(),
+        );
+        cb.add_volume_mount("kerberos", "/stackable/kerberos");
+
+        // Needed env vars
+        cb.add_env_var("KRB5_CONFIG", "/stackable/kerberos/krb5.conf");
+    }
+
+    Ok(())
+}
+
+pub fn kerberos_config_properties(
+    hive: &HiveCluster,
+    hive_namespace: &str,
+) -> BTreeMap<String, String> {
+    if !hive.has_kerberos_enabled() {
+        return BTreeMap::new();
+    }
+    let hive_name = hive.name_any();
+    let principal_host_part =
+        format!("{hive_name}.{hive_namespace}.svc.cluster.local@${{env.KERBEROS_REALM}}");
+
+    BTreeMap::from([
+        // Kerberos settings
+        (
+            "hive.metastore.kerberos.principal".to_string(),
+            format!(
+                "{service_name}/{principal_host_part}",
+                service_name = HiveRole::MetaStore.kerberos_service_name()
+            ),
+        ),
+        (
+            "hive.metastore.client.kerberos.principal".to_string(),
+            format!(
+                "{service_name}/{principal_host_part}",
+                service_name = HiveRole::MetaStore.kerberos_service_name()
+            ),
+        ),
+        (
+            "hive.metastore.kerberos.keytab.file".to_string(),
+            "/stackable/kerberos/keytab".to_string(),
+        ),
+        // Enabled https as well
+        (
+            "hive.metastore.sasl.enabled".to_string(),
+            "true".to_string(),
+        ),
+    ])
+}
+
+pub fn kerberos_container_start_commands(hive: &HiveCluster) -> String {
+    if !hive.has_kerberos_enabled() {
+        return String::new();
+    }
+
+    let mut args = vec![formatdoc! {"
+        export KERBEROS_REALM=$(grep -oP 'default_realm = \\K.*' /stackable/kerberos/krb5.conf)
+        sed -i -e 's/${{env.KERBEROS_REALM}}/'\"$KERBEROS_REALM/g\" {STACKABLE_CONFIG_DIR}/{HIVE_SITE_XML}",
+    }];
+
+    if hive.spec.cluster_config.hdfs.is_some() {
+        args.extend([
+            formatdoc! {"
+        sed -i -e 's/${{env.KERBEROS_REALM}}/'\"$KERBEROS_REALM/g\" {STACKABLE_CONFIG_DIR}/core-site.xml
+        sed -i -e 's/${{env.KERBEROS_REALM}}/'\"$KERBEROS_REALM/g\" {STACKABLE_CONFIG_DIR}/hdfs-site.xml",
+    }
```

Comment:
Nit: Some formatting issues

Reply:
Fixed in [c39b181](https://github.com/stackabletech/hive-operator/pull/402/commits/c39b181993d71d04536c75b912ab510bade600d4).